invenio/modules/access/api.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Provide API-callable functions for access management."""
+
+from intbitset import intbitset
+
+from invenio.base.globals import cfg
+from invenio.ext import principal
+from invenio.ext.sqlalchemy import db
+from invenio.modules.access import models
+from invenio.modules.access.local_config import DEF_AUTHS, DEF_ROLES, \
+    DEF_USERS, SUPERADMINROLE
+from invenio.modules.accounts.models import User
+
+from six import iteritems
+
+
+def get_action_roles(id_action):
+    """Return all the roles connected with an action."""
+    id_superadmin_role = models.AccROLE.factory(name=SUPERADMINROLE).id
+    return models.AccAuthorization.query.filter(db.or_(
+        models.AccAuthorization.id_accACTION == id_action,
+        models.AccAuthorization.id_accROLE == id_superadmin_role
+    )).distinct().all()
+
+
+def get_roles_emails(id_roles):
+    """Get emails by roles."""
+    return set(
+        map(lambda u: u.email.lower().strip(),
+            db.session.query(db.func.distinct(User.email)).join(
+                User.active_roles
+            ).filter(
+                models.UserAccROLE.id_accROLE.in_(id_roles)).all()))
+
+
+def find_possible_roles(name_action, always_add_superadmin=True,
+                        batch_args=False, **arguments):
+    """Find all the possible roles that are enabled to a given action.
+
+    :return: roles as a list of role_id
+    """
+    query_roles_without_args = \
+        db.select([models.AccAuthorization.id_accROLE], db.and_(
+            models.AccAuthorization.argumentlistid <= 0,
+            models.AccAuthorization.id_accACTION == db.bindparam('id_action')))
+
+    query_roles_with_args = \
+        models.AccAuthorization.query.filter(db.and_(
+            models.AccAuthorization.argumentlistid > 0,
+            models.AccAuthorization.id_accACTION == db.bindparam('id_action')
+        )).join(models.AccAuthorization.argument)
+
+    id_action = db.session.query(models.AccACTION.id).filter(
+        models.AccACTION.name == name_action).scalar()
+    roles = intbitset(db.engine.execute(query_roles_without_args.params(
+        id_action=id_action)).fetchall())
+
+    if always_add_superadmin:
```

Comment:
Yep! No worries, @hachreak. I just wanted to point this thing out for when it happens :dancer: